Fix/navbar users admin only

nikrb opened the pull request:
display users navbar menu item only for admin users
pass user role to FE on login
add BE check for admin role auth for users list

fixes #35

nikrb committed:
implement admin role BE and pass role on login

nikrb committed:
fix: route admin role to frontend

restrict menu users to admin only
route to login on users route if auth role not admin

nikrb committed:
test: fix users/getAll description

nikrb committed:
test: fix users/getAll

nikrb committed:
test: fix navbar

fix test for member user, users item should not be visible
add test for admin user, users item visible

nikrb committed:
fix: linting

jenovs commented on backend/routes/auth/isLoggedInAdmin.js:
```diff
@@ -0,0 +1,11 @@
+  if (
+    process.env.NODE_ENV === 'test' ||
+    (req.isAuthenticated() && req.user.role === 'admin')
+  ) {
```
I prefer doing something like this:
```js
const isTestEnv = process.env.NODE_ENV === 'test';
const isAuthenticated = req.isAuthenticated();
const hasAdminRole = req.user.role === 'admin';

if (isTestEnv || (isAuthenticated && hasAdminRole) {
...
}
```
It's a bit longer, but much more readable.

nikrb replied:
you have a point there, though I think the isAuthenticated is a bit OTT.

jenovs reviewed (approved):
Looks good, only BE tests are failing and FE tests warns about prop types :wink:

nikrb commented:
thanx for catching the test failure/warnings

nikrb committed:
test: add isAdmin to guest props in navbar test

jenovs commented:
BE tests are still failing

nikrb commented:
@jenovs I'm seeing a clean test run front and back. Can you give me some detail as to what is failing for you?

jenovs commented:
![be test error](https://user-images.githubusercontent.com/10571704/41496774-84c12b7e-7150-11e8-8593-99a98bf830b5.png)

nikrb commented:
hmmm I don't see that. can you try re-installing? I wasn't aware of any auth requirements in the test, but will have a look

nikrb committed:
review: tidy code

nikrb committed:
fix: clear admin flag on login fail

jenovs commented:
I think I found the problem: `backend/test/setup.js` line 4:
```js
require('dotenv').config({ path: '.env.test' });
```
README doesn't have anything about `.env.test`.

nikrb commented:
oops - sorry about that dude, I, of course, have a .env.test. Thanks for fixing it, I'll update readme.

I've made the review changes but found a bug with the navbar for admin role, which kinda opened a can of worms which I need to fix, so will probably need another review when I'z done.

nikrb committed:
tidy login flags

nikrb committed:
removed mocha options for backend test

removed the --reporter progress options. The bar is all very pretty
but without it we get to see the individual tests

nikrb committed:
test: fix BE isLoggedInAdmin

nikrb commented:
had to remove hasAdminRole as req.user is not populated during test
